### PR TITLE
fix: Fix CurrentStateChanged raised in incorrect time

### LINKED ISSUE #3160: VisualStateManager: `CurrentStateChanged` never raised for state with a `Forever` Storyboard
# Bug
## Current behavior
The event `CurrentStateChanging` is correctly raised, but the `CurrentStateChanged` is never raised for state defined in this way:
``` xml
<Grid x:Name="root">
  <VisualStateManager.VisualStateGroups>
    <VisualStateGroup x:Name="States">
      <VisualState x:Name="Normal" />
      <VisualState x:Name="Blinking">
        <Storyboard RepeatBehavior="Forever">
          [animations]
        </Storyboard>
      </VisualState>
    </VisualStateGroup>
  <VisualStateManager.VisualStateGroups>
</Grid>
```

``` csharp
  Loaded += (snd, evt) =>
  {
    var states = VisualStateManager.GetVisualStateGroups(root);
    states.First()
      .CurrentStateChanging += (s, e) => Console.Write($">>{e.NewState.Name}>>");
    states.First()
       // this is never reached for "Blinking" state
      .CurrentStateChanged += (s, e) => Console.Write($" [{e.NewState.Name}]\n");
  }
```

## Expected behavior
On UWP, the state is raised properly after first loop of the `Forever` animation.

**TO CHECK**: Maybe there's a similar bug when a `Forever` is set on a `<VisualTransition>`.

## Environment

Package Version(s): `2.3`

Affected platform(s):

- [X] iOS
- [X] Android
- [X] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [X] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #3160

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Youssef1313** [comment]
> C:\Program Files\dotnet\sdk\6.0.100-rc.1.21452.64\NuGet.targets(130,5): Error : Failed to download package 'System.Threading.ThreadPool.4.0.10' from 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/flat2/system.threading.threadpool/4.0.10/system.threading.threadpool.4.0.10.nupkg'.

networking error. Re-triggering a new build.

File: src/Uno.UI/UI/Xaml/VisualStateGroup.cs [hunk 1]
```diff
@@ -303,19 +303,10 @@ void ApplyTargetState()
...
-
-					void OnStateStoryboardCompleted(object s, object a)
-					{
-						state.Storyboard.Completed -= OnStateStoryboardCompleted;
-						onStateChanged();
-					}
-				}
-				else
-				{
-					onStateChanged();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/VisualStateGroup.cs:317]: /cc @dr1rrb does this relate to changes you're making ?

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/VisualStateGroup.cs:317]: This feels odd removing this part is not breaking anything else, maybe david will know more.

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/VisualStateGroup.cs:317]: @jeromelaban Yeah I have the same feeling. But I saw on UWP even for non-Forever animations, the state changed is fired much earlier than Storyboard's Completed. So it seems to make some sense, but I don't know why this code existed at the first place.

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/VisualStateGroup.cs:317]: @dr1rrb Would you be able to check? Thanks!

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/VisualStateGroup.cs:317]: I think this was made when the "transitions" concept was not applied properly, but I've tested and I confirm that the `CurrentStateChanged` event is raised **after** the transition but at the beginning of the state animation.

**Pull request merged**